src/CommerceContext.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Frontkom\DrupalBehatDefinitions;
+
+use Behat\Gherkin\Node\TableNode;
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+
+/**
+ * Class CommerceContext.
+ *
+ * Provide Behat step-definitions for common Commerce functionalities.
+ */
+class CommerceContext extends RawDrupalContext {
+
+  /**
+   * An array of promotions created during this context.
+   *
+   * @var array
+   */
+  protected $promotions = [];
+
+  /**
+   * Generate coupons.
+   *
+   * @Given coupons:
+   */
+  public function createCoupons(TableNode $nodesTable) {
+    $storage = \Drupal::entityTypeManager()->getStorage('commerce_promotion_coupon');
+    foreach ($nodesTable->getHash() as $nodeHash) {
+      $coupon = (object) $nodeHash;
+      $promotion = $this->getPromotionByName($coupon->promotion);
+      if ($promotion) {
+        $coupon_saved = $this->couponCreate($coupon);
+        $coupon_loaded = $storage->load($coupon_saved->id);
+        $promotion->get('coupons')->appendItem($coupon_loaded);
+        $promotion->save();
+      }
+      else {
+        throw new \Exception("No parent promotion found.");
+      }
+    }
+  }
+
+  /**
+   * Load promotion by name.
+   */
+  public function getPromotionByName($promotion_name) {
+    $storage = \Drupal::entityTypeManager()->getStorage('commerce_promotion');
+    foreach ($this->promotions as $promotion) {
+      $loaded_promotion = $storage->load($promotion->id);
+      if ($loaded_promotion->label() === $promotion_name) {
+        return $loaded_promotion;
+      }
+    }
+
+    return NULL;
+  }
+
+  /**
+   * Create coupon.
+   */
+  public function couponCreate($coupon) {
+    $saved = $this->getDriver()->createEntity('commerce_promotion_coupon', $coupon);
+    return $saved;
+  }
+
+  /**
+   * Visit promotion edit page.
+   *
+   * @Then I visit promotion :title edit page
+   */
+  public function iVisitPromotionEditPage($title) {
+    $promotions = \Drupal::entityTypeManager()->getStorage('commerce_promotion')
+      ->loadByProperties([
+        'name' => $title,
+      ]);
+    if (count($promotions) !== 1) {
+      throw new \Exception('Expected 1 promotion with title ' . $title . ' but found ' . count($promotions));
+    }
+
+    $id = reset($promotions)->id();
+    $this->getSession()->visit($this->locatePath("promotion/$id/edit"));
```

Comment:
An better way would be to use the entity directly and say

$promotion->toUrl('edit-form') or something like this. This way, if for some reason a site has a different path defined for the edit form (either through an override or different version of commerce) then it would work regardless